ui/src/components/Collection/CollectionItem/index.tsx

```diff
@@ -11,6 +11,13 @@ interface ICollectionItem {
 const CollectionItem = (props: ICollectionItem) => {
   const LibrariesCtx = useContext(LibrariesContext)
 
+    const renderDeleteAfterText = () => {
+      if (props.collection.deleteAfterDays === undefined) {
+        return "Never"
+      }
+      return `After ${props.collection.deleteAfterDays} days`
+    }
+
```

Comment:
I'd personally just keep all of this inline in the template as it's not very complicated.